subsys/net/lib/sockets/sockets_can.c

```diff
@@ -95,20 +95,23 @@ static void zcan_received_cb(struct net_context *ctx, struct net_pkt *pkt,
 	 * which will give the packet to correct net_context(s).
 	 */
 	struct net_pkt *clone = NULL;
+	struct zcan_frame *zframe = (struct zcan_frame *)net_pkt_data(pkt);
+	struct can_frame frame;
 	int i;
 
-	for (i = 0; i < ARRAY_SIZE(receivers); i++) {
-		struct zcan_frame *zframe =
-			(struct zcan_frame *)net_pkt_data(pkt);
-		struct can_frame frame;
+	if (net_pkt_get_len(pkt) != sizeof(struct zcan_frame)) {
+		NET_ERR("pkt size missmatch");
```

Comment:
Please avoid `NET_` prefixed macros in the samples, we have been trying to limit their use to core net code only. So better here to do `LOG_ERR`. The reason for this is that the sample code can be directly added to user application and user does not need to tweak the log entries or add any additional network specific config options.

Reply:
This is net code. It is in the sockets where everything is logged with NET_